tests/foreman/cli/test_subnet.py

```diff
@@ -581,3 +593,102 @@ def test_positive_delete_subnet_parameter_overrided_host_impact():
 
     :BZ: 1426612
     """
+
+
+def test_positive_subnet_set_parameters(module_target_sat, module_org, module_location):
+    """Create subnet, set parameters and delete subnet
+
+    :id: 17e039d3-2f7a-4d36-a952-47ca67514ca1
+
+    :steps:
+
+        1. Create valid subnet
+        2. Add valid parameter to subnet
+        3. Delete the above parameter from subnet
+        4. Delete subnet
+
+    :expectedresults: Subnet is created, set with parameters and deleted
+
+    """
+    name = gen_choice(list(valid_data_list().values()))
+    pool = sorted(valid_addr_pools()[0])
+    mask = '255.255.255.0'
+    # generate pool range from network address
+    network = gen_ipaddr()
+    from_ip = re.sub(r'\d+$', str(pool[0]), network)
+    to_ip = re.sub(r'\d+$', str(pool[1]), network)
+    domains_amount = random.randint(2, 3)
+    domains = [
+        module_target_sat.cli_factory.make_domain(
+            {'location-ids': module_location.id, 'organization-ids': module_org.id}
+        )
+        for _ in range(domains_amount)
+    ]
+    gateway = gen_ipaddr(ip3=True)
+    ipam_type = SUBNET_IPAM_TYPES['dhcp']
+
+    # Create subnet
+    subnet = module_target_sat.cli_factory.make_subnet(
+        {
+            'name': name,
+            'from': from_ip,
+            'mask': mask,
+            'network': network,
+            'to': to_ip,
+            'domain-ids': [domain['id'] for domain in domains],
+            'gateway': gateway,
+            'ipam': ipam_type,
+            'organization-id': module_org.id,
+            'location-id': module_location.id,
+        }
+    )
+    # Check if Subnet can be listed
+    subnets_ids = [subnet_['id'] for subnet_ in module_target_sat.cli.Subnet.list()]
+    assert subnet['id'] in subnets_ids
+    assert subnet['name'] == name
+    assert subnet['start-of-ip-range'] == from_ip
+    assert subnet['end-of-ip-range'] == to_ip
+    assert len(subnet['domains']) == domains_amount
+    for domain in domains:
+        assert domain['name'] in subnet['domains']
+    assert gateway in subnet['gateway-addr']
+    assert ipam_type in subnet['ipam']
+
+    name = gen_string('alphanumeric', 10)
+    value = gen_string('alphanumeric', 10)
+    # Set parameters to subnet
+    module_target_sat.cli.Subnet.set_parameter(
+        {'subnet-id': subnet['id'], 'name': name, 'value': value}
+    )
+    # Fetch subnet info
+    subnet = module_target_sat.cli.Subnet.info({'id': subnet['id']}, output_format='json')
```

Comment:
I think default is json format, so no need to specify explicitly
```suggestion
    subnet = module_target_sat.cli.Subnet.info({'id': subnet['id']})
```

Reply:
Using output_format='json' preserves structure, While without using output format it Flattens structure i.e Converts hierarchical text into flat dictionary.
As i have to access nested data structures (like parameter); using output_format='json'.

Reply:
I agree that using `output_format='json'` provides more detailed information about the parameters, but using `csv` also provides the required information. Providing the suggestions below